src/lib/device/hs_super_blk.h

```diff
@@ -0,0 +1,189 @@
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include <iomgr/iomgr.hpp>
+#include <sisl/fds/sparse_vector.hpp>
+#include <homestore/homestore_decl.hpp>
+
+namespace homestore {
+
+#pragma pack(1)
+struct disk_attr {
+    // all fields in this structure are a copy from iomgr::drive_attributes;
+    uint32_t phys_page_size{0};        // Physical page size of flash ssd/nvme. This is optimal size to do IO
+    uint32_t align_size{0};            // size alignment supported by drives/kernel
+    uint32_t atomic_phys_page_size{0}; // atomic page size of the drive_sync_write_count
+    uint32_t num_streams{0};
+
+    disk_attr() = default;
+    disk_attr(const iomgr::drive_attributes& iomgr_attr) :
+            phys_page_size{iomgr_attr.phys_page_size},
+            align_size{iomgr_attr.align_size},
+            atomic_phys_page_size{iomgr_attr.atomic_phys_page_size},
+            num_streams{iomgr_attr.num_streams} {}
+
+    disk_attr& operator=(const iomgr::drive_attributes& iomgr_attr) {
+        phys_page_size = iomgr_attr.phys_page_size;
+        align_size = iomgr_attr.align_size;
+        atomic_phys_page_size = iomgr_attr.atomic_phys_page_size;
+        num_streams = iomgr_attr.num_streams;
+
+        return *this;
+    }
+
+    bool is_valid() const {
+        return is_page_valid(phys_page_size) && is_page_valid(align_size) && is_page_valid(atomic_phys_page_size);
+    }
+
+    bool is_page_valid(uint32_t page_size) const {
+        return (page_size == 0 || (page_size & (page_size - 1)) != 0) ? false : true;
+    }
+
+    std::string to_string() const {
+        return fmt::format("phys_page_size={}, align_size={}, atomic_phys_page_size={}, num_streams={}",
+                           in_bytes(phys_page_size), in_bytes(align_size), in_bytes(atomic_phys_page_size),
+                           num_streams);
+    }
+};
+
+struct first_block_header {
+    static constexpr const char* PRODUCT_NAME{"OmStore"};
+    static constexpr size_t s_product_name_size{64};
+    static constexpr uint32_t CURRENT_SUPERBLOCK_VERSION{4};
+
+public:
+    uint64_t gen_number{0};                   // Generation count of this structure
+    uint32_t version{0};                      // Version Id of this structure
```

Comment:
Should it be initialized to 'CURRENT_SUPERBLOCK_VERSION'?

Reply:
nevermind, after loading the diff, I can it it is initialized in format_devices, which is the right way to do.